src/data/slides.tsx

```diff
@@ -0,0 +1,261 @@
+        slide.isMultiplexedSamples = true
+      } else if (slide.areImagesMonochrome) {
+        slide.description = 'Monochrome Slide'
+      } else {
+        slide.description = 'RGB Slide'
+      }
+    }
+  }
+  return slides
+}
+
+/**
+ * Parses volume instances into a slide.
+ *
+ * @params volumeMetadataList - array of volume instances
+ * @params slide
+ * @returns volumeInstanceReference - first volume instance of the list
+ */
+function parseVolumeMetadataFromListToSlide (
+  volumeMetadataList: object[],
+  slide: Slide
+): any {
+  let volumeInstanceReference
+  for (let j = 0; j < volumeMetadataList.length; ++j) {
+    const metadata = volumeMetadataList[j]
+    const instance =
+      dmv.metadata.formatMetadata(metadata) as dmv.metadata.VLWholeSlideMicroscopyImage
+    if (j === 0) {
+      slide.frameofReferenceUID = instance.FrameOfReferenceUID
+    } else if (slide.frameofReferenceUID !== instance.FrameOfReferenceUID) {
+      console.warn('FrameOfReferenceUID of volume instance' +
+                   instance.SOPInstanceUID +
+                   ' does not correspond to slide FrameOfReferenceUID. ' +
+                   'The instance will be discarded.')
+      continue
+    }
+    if (j === 0) {
+      slide.containerIdentifier = instance.ContainerIdentifier
+    } else if (slide.containerIdentifier !== instance.ContainerIdentifier) {
+      console.warn('ContainerIdentifier of volume instance' +
+                   instance.SOPInstanceUID +
+                   ' does not correspond to slide ContainerIdentifier. ' +
+                   'The instance will be discarded.')
+      continue
+    }
+    const instanceOpticalPathIdentifier =
+      instance.OpticalPathSequence[0].OpticalPathIdentifier
+    const instanceIsMonochorme = instance.SamplesPerPixel === 1 &&
+      instance.PhotometricInterpretation === 'MONOCHROME2'
+    if (slide.keyOpticalPathIdentifier === '' && instanceIsMonochorme) {
+      slide.areImagesMonochrome = true
+      slide.keyOpticalPathIdentifier = instanceOpticalPathIdentifier
+    } else if (instanceIsMonochorme !== slide.areImagesMonochrome) {
+      console.warn('Volume instance' +
+                   instance.SOPInstanceUID +
+                   ' of the slide has different image type. ' +
+                   'The instance will be discarded.')
+      continue
+    }
+    if (volumeInstanceReference === undefined) {
+      volumeInstanceReference = instance
+    }
+    if (slide.opticalPathIdentifiersList.findIndex(
+      (opi) => opi === instanceOpticalPathIdentifier) === -1
+    ) {
+      slide.opticalPathIdentifiersList.push(instanceOpticalPathIdentifier)
+    }
+    slide.volumeMetadata.push(metadata)
+  }
+  return volumeInstanceReference
+}
+
+/**
+ * Parses label instances into a slide.
+ *
+ * @params labelMetadataList - array of label instances
+ * @params slide
+ */
+function parseLabelMetadataFromListToSlide (
+  labelMetadataList: object[],
+  slide: Slide
```

Comment:
I would suggest passing the relevant attributes directly

```suggestion
  frameOfReferenceUID: string,
  containerIdentifier: string
```

Same for the other functions.

Reply:
done https://github.com/MGHComputationalPathology/slim/pull/25/commits/b457fc27cee8440c76bb315a09b78f4b05a20732